src/logging/index.ts

```diff
@@ -0,0 +1,62 @@
+import { Disposable, LogLevel, LogOutputChannel, window } from "vscode";
+import { extendErrorWithSupportLinks, provideSingleton } from "../utils";
+import { TelemetryService } from "../telemetry";
+import { CustomException } from "./exception";
+
+@provideSingleton(LoggingService)
+export class LoggingService implements Disposable {
+  private disposables: Disposable[] = [];
+  private outputChannel: LogOutputChannel | null = null;
+  // @ts-ignore
+  private logLevel: LogLevel;
+  constructor(private telemetry: TelemetryService) {
+    this.outputChannel = window.createOutputChannel("DBTPowerUser", {
+      log: true,
+    });
+    this.outputChannel.onDidChangeLogLevel((e) => {
+      this.logLevel = e;
+    });
+    this.disposables.push(Disposable.from(this.outputChannel));
+  }
+
+  trace(msg: string) {
+    this.outputChannel?.appendLine(msg);
+    console.log(msg);
+  }
+
+  debug(msg: string) {
+    this.outputChannel?.debug(msg);
+    console.debug(msg);
+  }
+
+  info(e: CustomException) {
+    const errorMessage = e.getErrorMessage();
+    this.outputChannel?.info(errorMessage);
+    this.telemetry.sendTelemetryEvent(e.name, e.extra);
+    console.info(errorMessage);
+  }
+
+  warn(e: CustomException) {
+    const errorMessage = e.getErrorMessage();
+    this.outputChannel?.warn(errorMessage);
+    this.telemetry.sendTelemetryError(e.name, e.error, e.extra);
+    console.warn(errorMessage);
+  }
+
+  error(e: CustomException) {
+    const errorMessage = e.getErrorMessage();
+    this.outputChannel?.error(errorMessage);
+    this.telemetry.sendTelemetryError(e.name, e.error, e.extra);
+    window.showErrorMessage(extendErrorWithSupportLinks(e.message));
+    console.error(errorMessage);
+  }
```

Comment:
I'm not sure about the bundling with telemetry. this might increase heavily the amount of events.